Dockerfile

```
@@ -1,22 +1,20 @@
 FROM ubuntu:20.04
 
-ENV SMB_USER samba
-ENV SMB_PASS password
+ARG SMB_USER=samba
+ARG SMB_PASS=password
 
-RUN apt-get clean && \
-    apt-get update && \
-    apt-get upgrade -y && \
-    apt-get install samba=2:4.15.13+dfsg-0ubuntu0.20.04.2 smbclient=2:4.15.13+dfsg-0ubuntu0.20.04.2 -y && \
-    rm -fr /var/cache/*
+RUN apt-get update && \
+    apt-get install --no-install-recommends samba=2:4.15.13+dfsg-0ubuntu0.20.04.2 smbclient=2:4.15.13+dfsg-0ubuntu0.20.04.2 -y && \
+    apt-get clean
```

Let's shrink the image a bit further
```suggestion
    apt-get clean && \
    rm -rf /var/lib/apt/lists/*
```